patches/components-safe_browsing-browser-BUILD.gn.patch

```diff
@@ -0,0 +1,12 @@
+diff --git a/components/safe_browsing/browser/BUILD.gn b/components/safe_browsing/browser/BUILD.gn
+index 777d390eeaf4f0960c7f2a0f9c3fecddcc46d12f..a995660063df062cf0e01b647a26ba67c717e947 100644
+--- a/components/safe_browsing/browser/BUILD.gn
++++ b/components/safe_browsing/browser/BUILD.gn
+@@ -52,6 +52,7 @@ source_set("network_context") {
+     "//content/public/browser:browser",
+     "//net:extras",
+   ]
++  deps += [ "//brave/services/network/public/cpp" ]
```

Comment:
@bridiver We currently patched network::SystemRequestHandler::GetInstance()->OnBeforeSystemRequest(request) in `safe_browsing_network_context.cc` without adding its direct deps here  and I encountered undefined symbol error due to this, so I patched here.
If there is a better way to get the deps in without patching here, please suggest, thank you.

Reply:
we should always patch in a generic dep for the component, not a specific dep that it needs because this is not extensible. So this should be `//brave/components/safe_browsing/browser`

Reply:
actually you shouldn't need this patch at all, this is already included in `//services/network/public/cpp`, but that dep is missing in upstream for `network_context` so we should fix and submit to upstream

Reply:
As discussed in DM, currently it is enough for us to just make brave a public dep in `//services/network/public/cpp` for now.
Will submit a patch to upstream since `network_context` should add a dep of `//services/network/public/cpp`.